src/screens/registerStudent/hooks/useRegister.ts

```diff
@@ -0,0 +1,248 @@
+  const [enrollment, setEnrollment] = useState('');
+  const [description, setDescription] = useState('');
+  const [email, setEmail] = useState('');
+  const [contactEmail, setContactEmail] = useState('');
+  const [linkedin, setLinkedin] = useState('');
+  const [whatsapp, setWhatsapp] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [nameError, setNameError] = useState('');
+  const [enrollmentError, setEnrollmentError] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const handleContinueClick = (e: React.SyntheticEvent<EventTarget>) => {
+    let currentError;
+
+    e.preventDefault();
+    setNameError('');
+    setConfirmPasswordError('');
+    setPasswordError('');
+    setEmailError('');
+
+    currentError = validateName(name);
+    if (currentError !== '') {
+      return setNameError(currentError);
+    }
+
+    currentError = validateEmail(email);
+    if (currentError !== '') {
+      return setEmailError(currentError);
+    }
+
+    currentError = validatePassword(name, password);
+    if (currentError !== '') {
+      return setPasswordError(currentError);
+    }
+
+    currentError = validateConfirmPassword(password, confirmPassword);
+    if (currentError !== '') {
+      return setConfirmPasswordError(currentError);
+    }
+
+    handleStepChange(1);
+  };
+
+  const handleStepChange = (increment: number) => {
+    setStep((previousStep) => previousStep + increment);
+  };
+
+  const handleBackClick = () => {
+    handleStepChange(-1);
+  };
+
+  const handleSecondContinueClick = (e: React.SyntheticEvent<EventTarget>) => {
+    e.preventDefault();
+    setEnrollmentError('');
+
+    const currentError = validateEnrollment(enrollment);
+    if (currentError !== '') {
+      return setEnrollmentError(currentError);
+    }
+
+    handleStepChange(1);
+  };
+
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const handleMouseDownPassword = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
+
+  const handleContactEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setContactEmail(e.target.value);
+  };
+
+  const handleLinkedinChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLinkedin(e.target.value);
+  };
+
+  const handleWhatsappChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setWhatsapp(e.target.value);
+  };
+
+  const handleCourseChange = (e: SelectChangeEvent<unknown>) => {
+    const value = e.target.value as string;
+    const code = value.split(' - ')[0];
+    let id = 0;
+
+    courses.forEach((course) => {
+      if (course.code === code) {
+        id = course.id;
+        return;
+      }
+    });
```

Comment:
Ao invés de `forEach` aqui você devia usar o método `find`. Segura **[aqui](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)** a referência.
```suggestion
    const course = courses.find(course => course.code === code);
    const id = course?.id;
```

Reply:
Bixa eu sabia que existia um jeito typescriptonico de fazer isso mas eu já tava viciado no código deixei de qualquer forma. De qualquer jeito obrigado amiga, aderido